src/bootstrap/lib.rs

```diff
@@ -1242,44 +1242,89 @@ impl Build {
         if self.config.dry_run() {
             return Some(PathBuf::new());
         }
+
         if let Some(linker) = self.config.target_config.get(&target).and_then(|c| c.linker.clone())
         {
             Some(linker)
         } else if target.contains("vxworks") {
             // need to use CXX compiler as linker to resolve the exception functions
             // that are only existed in CXX libraries
             Some(self.cxx.borrow()[&target].path().into())
-        } else if target != self.config.build
-            && util::use_host_linker(target)
-            && !target.contains("msvc")
+        } else if target != self.config.build && util::use_host_linker(target) && !target.is_msvc()
         {
             Some(self.cc(target))
-        } else if self.config.use_lld && !self.is_fuse_ld_lld(target) && self.build == target {
-            Some(self.initial_lld.clone())
+        } else if self.config.lld_mode.is_used()
+            && self.is_lld_direct_linker(target)
+            && self.build == target
+        {
+            match self.config.lld_mode {
+                LldMode::SelfContained => Some(self.initial_lld.clone()),
+                LldMode::External => Some("lld".into()),
```

Comment:
Where is the `-flavor` option passed to LLD?
The `lld` and `rust-lld` binaries are not functional by themselves, you get something like this from them
```rust
$ lld
lld is a generic driver.
Invoke ld.lld (Unix), ld64.lld (macOS), lld-link (Windows), wasm-ld (WebAssembly) instead
```

Reply:
I suppose that for MSVC it works just by setting the linker, without setting any specific flavors? It has worked like this before this PR, for MSVC only the linker was changed, and no other LLD flags were set (apart of the 1 thread flag for tests).

Reply:
This is no longer relevant after the switch to MCP510.

Reply:
Ah, wait, it's still relevant to MSCV.
MSCV should probably use MCP510 too.